packages/extension-sdk/src/connect/extension_host_api.ts

```diff
@@ -52,54 +53,73 @@ export class ExtensionHostApiImpl implements ExtensionHostApi {
   private chattyHost: ChattyHostConnection
   private setInitialRoute?: (route: string, routeState?: any) => void
   private hostChangedRoute?: (route: string, routeState?: any) => void
+  private vizualizationDataReceivedCallback?: VizualizationDataReceivedCallback
+
   private contextData?: string
 
   constructor(configuration: ExtensionHostApiConfiguration) {
     this._configuration = configuration
-    const { chattyHost, setInitialRoute, hostChangedRoute } =
-      this._configuration
+    const {
+      chattyHost,
+      setInitialRoute,
+      hostChangedRoute,
+      vizualizationDataReceivedCallback,
+    } = this._configuration
     this.chattyHost = chattyHost
     this.setInitialRoute = setInitialRoute
     this.hostChangedRoute = hostChangedRoute
+    this.vizualizationDataReceivedCallback = vizualizationDataReceivedCallback
   }
 
   get lookerHostData() {
     return this._lookerHostData
   }
 
   handleNotification(
-    message?: ExtensionNotification
+    message: ExtensionNotification
   ): ExtensionInitializationResponse | undefined {
-    const { type, payload } = message || {}
+    const { type } = message
     switch (type) {
-      case ExtensionNotificationType.ROUTE_CHANGED:
+      case ExtensionNotificationType.ROUTE_CHANGED: {
+        const { payload } = message
         if (this.hostChangedRoute && payload) {
-          const { route, routeState } = payload as RouteChangeData
+          const { route, routeState } = payload
           if (route) {
             this.hostChangedRoute(route, routeState)
           }
         }
         return undefined
+      }
+      case ExtensionNotificationType.VISUALIZATION_DATA: {
+        const { payload } = message
+        if (this.vizualizationDataReceivedCallback) {
+          this.vizualizationDataReceivedCallback(payload)
+        }
+        return undefined
+      }
       case ExtensionNotificationType.INITIALIZE: {
-        this._lookerHostData = payload as LookerHostData
-        if (this._lookerHostData) {
-          this.contextData = this._lookerHostData.contextData
+        const { payload } = message
+        const lookerHostData = payload || {}
+        if (!lookerHostData.mountPoint) {
+          lookerHostData.mountPoint = MountPoint.standalone
         }
+        this._lookerHostData = lookerHostData
+        this.contextData = lookerHostData.contextData
         let errorMessage
         if (
           this._configuration.requiredLookerVersion &&
-          this._lookerHostData &&
-          this._lookerHostData.lookerVersion
+          lookerHostData.lookerVersion
         ) {
           errorMessage = this.verifyLookerVersion(
             this._configuration.requiredLookerVersion
           )
           if (errorMessage) {
+            // eslint-disable-next-line no-console
             console.error(errorMessage)
```

Comment:
do we use this convention in multiple places? would it be worth creating a trivial function for console errors instead and decorating the lint override in one place?